### PR TITLE
fix: display all class meet times

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Pre-flight checklist

- [x] I have searched for potential duplicate pull requests.
- [x] **If this is a code change**: I have written unit tests and/or conducted manual tests to fully verify the new behavior.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes https://feedback.coursetable.com/bugs/p/see-all-meeting-times-for-a-class) and the maintainers have approved on my working plan.

## Summary

This PR makes all times and locations considered by:

1. Conflict checking
2. Calendar view
3. Export ICS/GCal
4. Course modal

## Test plan (required)

Before:

image

After:

image

Before:

image

After:

image

### CODE REVIEW

**Comment by bearsyankees** [comment]
sweet. @benzuzu @BradleyLewis08 may need to integrate this w gcal now... thoughts

**Comment by Josh-Cena** [comment]
I have no idea why Prettier works locally but not in CI...

**Comment by Josh-Cena** [comment]
I'm going to bump dependencies. We can worry about Prettier later. @bearsyankees We can get it merged tonight right?

File: frontend/src/components/CourseModal/CourseModalOverview.jsx [hunk 1]
```diff
@@ -81,12 +82,23 @@ function CourseModalOverview({ setFilter, filter, setSeason, listing }) {
...
-  // Parse for location url and location name
-  let location_url = '';
-  for (const i in weekdays) {
-    const day = weekdays[i];
-    if (listing.times_by_day && listing.times_by_day[day]) {
-      location_url = listing.times_by_day[day][0][3];
+  const locations = new Map();
+  const times = new Map();
+  for (const day of weekdays) {
+    const info = listing.times_by_day[day];
```
Comment (bearsyankees) [inline frontend/src/components/CourseModal/CourseModalOverview.jsx:88]: looks solid but something here makes it military time i think....

Comment (Josh-Cena) [reply frontend/src/components/CourseModal/CourseModalOverview.jsx:88]: @bearsyankees It's right: the time is now 24h. I'm not sure if that's something worth fixing as I'm personally more accustomed to 24h. If you think it's a bug, I can fix that later.

**Pull request merged**